Simcoe/mParticle/mParticle.swift

```diff
@@ -38,32 +38,74 @@ public class mParticle {
                                                 proxyAppDelegate: proxyAppDelegate)
     }
 
-    private func mPProduct(name: String, productId: String, quantity: Int, price: NSNumber?, additionalProperties: Properties?) -> MPProduct {
+    private func mPProduct(name: String,
+                           productId: String,
+                           quantity: Int,
+                           price: NSNumber?,
+                           inout additionalProperties: Properties?) -> MPProduct {
         let product = MPProduct(name:name, sku:productId, quantity:quantity, price:price)
 
         if let brand = additionalProperties?[MPProductKeys.brand.rawValue] as? String {
             product.brand = brand
+            additionalProperties?.removeValueForKey(MPProductKeys.brand.rawValue)
         }
 
         if let category = additionalProperties?[MPProductKeys.category.rawValue] as? String {
             product.category = category
+            additionalProperties?.removeValueForKey(MPProductKeys.category.rawValue)
         }
 
         if let couponCode = additionalProperties?[MPProductKeys.couponCode.rawValue] as? String {
             product.couponCode = couponCode
+            additionalProperties?.removeValueForKey(MPProductKeys.couponCode.rawValue)
         }
 
         if let sku = additionalProperties?[MPProductKeys.sku.rawValue] as? String {
             product.variant = sku
+            additionalProperties?.removeValueForKey(MPProductKeys.sku.rawValue)
         }
 
         if let position = additionalProperties?[MPProductKeys.position.rawValue] as? UInt {
             product.position = position
+            additionalProperties?.removeValueForKey(MPProductKeys.position.rawValue)
         }
 
         return product
     }
 
+    private func mPCommerceEvent(eventType: MPCommerceEventAction,
+                                 product: MPProduct,
+                                 inout additionalProperties: Properties?) -> MPCommerceEvent {
```

Comment:
Would it make sense to create a separate dictionary for the purpose of passing SDK-specific properties?